### PR TITLE
Aurora pipeline finished : Caleb

### LINKED ISSUE #40: Make Pipeline Lambda handler for Quadhoral aurorawatch
As a developer, I want to create a Lambda handler script that orchestrates the aurora pipeline (extract, transform, and load processes), so the entire pipeline can run serverlessly and automatically in response to events, ensuring that updated aurora data is always available for amateur stargazers.

- Create pipeline script
- Write so that it conforms to AWS lambda conventions

### LINKED ISSUE #41: Dockerise Aurorawatch Quadhoral Pipeline
As a developer, I want the entire stargazing pipeline to be containerized using Docker, so I can ensure consistent environments across all deployments, making the app easier to run and deploy.

- Create `Dockerfile` for pipeline
- Build docker image
- Push image to ECR

### LINKED ISSUE #45: Create Requirements.txt for Quadhoral Aurorawatch Pipeline
As a developer, when I create the folder for the quadhoral pipeline, I want here to be an adjacent `requirements.txt` file so that the required dependencies can be easily installed both when working locally or when executing github workflows, so that they do not require manual installation which would slow progress.

- Create `requirements.txt`
- Update dependencies

### PR DESCRIPTION
**Pull request by CalebMathers**
This closes #41 
This closes #40 
This closes #45 

Made the Dockerfile and lambda handler for the entire pipeline, ready to be uploaded as an image to ECR

### CODE REVIEW

File: aurorawatch-pipeline/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+
+COPY requirements.txt .
+RUN pip install -r requirements.txt 
+
+COPY aurora_load.py .
+COPY aurora_extract.py .
+COPY aurora_transform.py .
+COPY aurora_pipeline.py .
+
+CMD [ "aurora_pipeline.lambda_handler" ]
```
Comment (AlexOBarnes) [inline aurorawatch-pipeline/Dockerfile:13]: Boilerplate lambda I approve

File: aurorawatch-pipeline/aurora_pipeline.py [hunk 2]
```diff
@@ -0,0 +1,23 @@
+'''Orchestrates weekly ETL pipeline'''
+# pylint: disable=W0612,W0613,E0606
+import logging
+
```
Comment (AlexOBarnes) [inline aurorawatch-pipeline/aurora_pipeline.py:4]: Is this linebreak necessary?

File: aurorawatch-pipeline/aurora_pipeline.py [hunk 3]
```diff
@@ -0,0 +1,23 @@
+'''Orchestrates weekly ETL pipeline'''
+# pylint: disable=W0612,W0613,E0606
```
Comment (AlexOBarnes) [inline aurorawatch-pipeline/aurora_pipeline.py:2]: Wow disabling linting, tut, tut, tut

File: aurorawatch-pipeline/Dockerfile [hunk 4]
```diff
@@ -0,0 +1,13 @@
+FROM public.ecr.aws/lambda/python:latest
+
+WORKDIR ${LAMBDA_TASK_ROOT}
```
Comment (AlexOBarnes) [inline aurorawatch-pipeline/Dockerfile:3]: Very necessary

**Review by AlexOBarnes — APPROVED** [review]
Very boilerplate looking code

File: aurorawatch-pipeline/Dockerfile (hunk 4 above)
Comment (olliejth) [reply aurorawatch-pipeline/Dockerfile:3]: Consistent with other folders

File: aurorawatch-pipeline/aurora_pipeline.py [hunk 5]
```diff
@@ -0,0 +1,23 @@
...
+
+    if data:
+        clean_data = transform(data)
+
+    if clean_data:
+        load_data(clean_data)
+
+
+if __name__ == '__main__':
+    lambda_handler({}, {})
```
Comment (olliejth) [inline aurorawatch-pipeline/aurora_pipeline.py:23]: Consistent naming convention.

File: aurorawatch-pipeline/aurora_pipeline.py
Comment (olliejth) [inline aurorawatch-pipeline/aurora_pipeline.py:1]: Could add a space between these lines?

**Review by olliejth — APPROVED** [review]
SOme minor formatting points but otherwise no issues.

**Pull request merged**